### PR TITLE
Add MAC address validator

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Validator for MAC addresses.

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
@ronanguilloux can you please restart this job ? https://travis-ci.org/ronanguilloux/IsoCodes/jobs/65304021

Fail is not related to my PR.

**Comment by Soullivaneuh** [comment]
@ronanguilloux Job failed again: https://travis-ci.org/ronanguilloux/IsoCodes/jobs/65681138#L244

It's always on php 5.4. Don't know why, any idea?

**Comment by ronanguilloux** [comment]
`255` code you find in [the Travis job trace](https://travis-ci.org/ronanguilloux/IsoCodes/jobs/65681138) is related to a [PHP exit status](http://php.net/manual/fr/function.exit.php), 
caused by more possible reasons:
- error_reporting is not defined and php reports no error at all
- An @ (error suppression operator) hides the output of the error
- STDERR is redirected somewhere else (php -f somefile.php 2>/dev/null, remove the redirection)

To debug your code, you may want to set `display_errors` to true:

```xml
<phpunit ....... >
    ...
    ...
    <php>
         <ini name="display_errors" value="true"/>
    </php>
</phpunit>
```

**Comment by Soullivaneuh** [comment]
> error_reporting is not defined and php reports no error at all

How this could be possible on Travis?

> An @ (error suppression operator) hides the output of the error

Didn't find any `@` on the code.

> STDERR is redirected somewhere else (php -f somefile.php 2>/dev/null, remove the redirection)

How it's possible? The only run command is phpunit AFAIK.

To debug your code, you may want to set `display_errors` to true:

I'll try on this PR. If this works, it should be integrated too.

**Comment by ronanguilloux** [comment]
This may help you: http://docs.travis-ci.com/user/languages/php/

**Comment by Soullivaneuh** [comment]
Well, this is not very better: https://travis-ci.org/ronanguilloux/IsoCodes/jobs/67896053#L837

I don't see what could help on this doc. Very weird... any idea?

**Comment by peter-mw** [comment]
it says 

```
.
Deprecated: Isbn10::validate validator is deprecated since 1.2 and will be removed in 2.0. Use Isbn::validate($value, 10) instead. in /home/travis/build/ronanguilloux/IsoCodes/src/IsoCodes/Isbn10.php on line 19
Call Stack:
    0.0003     240112   1. {main}() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/phpunit:0
    0.0066     749728   2. PHPUnit_TextUI_Command::main() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/phpunit:36
    0.0067     756312   3. PHPUnit_TextUI_Command->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/TextUI/Command.php:105
    0.3269   10842952   4. PHPUnit_TextUI_TestRunner->doRun() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/TextUI/Command.php:153
    0.3499   11082392   5. PHPUnit_Framework_TestSuite->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/TextUI/TestRunner.php:406
    3.8933   13837112   6. PHPUnit_Framework_TestSuite->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestSuite.php:722
    3.8965   14101864   7. PHPUnit_Framework_TestSuite->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestSuite.php:722
    3.9318   14429464   8. PHPUnit_Framework_TestCase->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestSuite.php:722
    3.9319   14429416   9. PHPUnit_Framework_TestResult->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestCase.php:699
    3.9322   14430456  10. PHPUnit_Framework_TestCase->runBare() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestResult.php:609
    3.9332   14447104  11. PHPUnit_Framework_TestCase->runTest() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestCase.php:743
    3.9333   14448064  12. ReflectionMethod->invokeArgs() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestCase.php:866
    3.9333   14448096  13. IsoCodes\Tests\Isbn10Test->testValidIsbn10() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestCase.php:866
    3.9333   14448096  14. IsoCodes\Isbn10::validate() /home/travis/build/ronanguilloux/IsoCodes/tests/IsoCodes/Tests/Isbn10Test.php:65
    3.9334   14448376  15. trigger_error() /home/travis/build/ronanguilloux/IsoCodes/src/IsoCodes/Isbn10.php:19
```

maybe this is the problem?

**Comment by Soullivaneuh** [comment]
@peter-mw No, just a deprecated warning. You can see it on another passing test.

**Comment by Soullivaneuh** [comment]
Found why:

```
PHP Fatal error:  Call to undefined function IsoCodes\boolval() in /home/travis/build/ronanguilloux/IsoCodes/src/IsoCodes/Mac.php on line 29
PHP Stack trace:
PHP   1. {main}() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/phpunit:0
PHP   2. PHPUnit_TextUI_Command::main() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/phpunit:36
PHP   3. PHPUnit_TextUI_Command->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/TextUI/Command.php:105
PHP   4. PHPUnit_TextUI_TestRunner->doRun() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/TextUI/Command.php:153
PHP   5. PHPUnit_Framework_TestSuite->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/TextUI/TestRunner.php:406
PHP   6. PHPUnit_Framework_TestSuite->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestSuite.php:722
PHP   7. PHPUnit_Framework_TestSuite->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestSuite.php:722
PHP   8. PHPUnit_Framework_TestCase->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestSuite.php:722
PHP   9. PHPUnit_Framework_TestResult->run() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestCase.php:699
PHP  10. PHPUnit_Framework_TestCase->runBare() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestResult.php:609
PHP  11. PHPUnit_Framework_TestCase->runTest() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestCase.php:743
PHP  12. ReflectionMethod->invokeArgs() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestCase.php:866
PHP  13. IsoCodes\Tests\MacTest->testValidMacAddress() /home/travis/build/ronanguilloux/IsoCodes/vendor/phpunit/phpunit/src/Framework/TestCase.php:866
PHP  14. IsoCodes\Mac::validate() /home/travis/build/ronanguilloux/IsoCodes/tests/IsoCodes/Tests/MacTest.php:55
```

**Comment by Soullivaneuh** [comment]
http://php.net/manual/fr/function.boolval.php

> (PHP 5 >= 5.5.0, PHP 7)

Well. :-D

**Comment by Soullivaneuh** [comment]
Should be OK now.

**Comment by Soullivaneuh** [comment]
Tests passes, ready to merge.

Don't know for coverage, test covers all `Mac` class.

BTW: https://coveralls.io/builds/2877289/source?filename=IsoCodes%2FMac.php

> SOURCE NOT AVAILABLE
The file "IsoCodes/Mac.php" isn't available on GitHub. Either it's been removed, or the repo root directory needs to be updated.

This should be fixed IMO.

**Comment by Soullivaneuh** [comment]
ping @ronanguilloux :-)

File: src/IsoCodes/Mac.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+     *
+     * @param string $mac
+     *
+     * @return bool
+     */
+    public static function validate($mac)
+    {
+        $pattern = '/^(([a-f0-9]{2}-){5}[a-f0-9]{2}|([A-F0-9]{2}-){5}[A-Z0-9]{2}|([a-f0-9]{2}:){5}[a-z0-9]{2}|([A-F0-9]{2}:){5}[A-Z0-9]{2})$/';
+
+        return boolval(preg_match($pattern, $mac));
```
Comment (ronanguilloux) [inline src/IsoCodes/Mac.php]: a single `boolval()` is causing many verbose consequences (2 more files, one of them is temporary, etc.).
To make simpler and easier, I'd suggest to use `(bool) $val` instead.

**Comment by Soullivaneuh** [comment]
@ronanguilloux review corrected.

**Comment by Soullivaneuh** [comment]
Test passed. Ready to merge.

ping @ronanguilloux

**Comment by Soullivaneuh** [comment]
ping @ronanguilloux 

Do you have some ETA to update this bundle?

**Comment by ronanguilloux** [comment]
Next october or november '15

**Comment by Soullivaneuh** [comment]
Thanks for the date.

**Comment by Soullivaneuh** [comment]
@ronanguilloux Going into news.

Did you get some time to work on this project? Need Help? Thanks! :-)

**Comment by Soullivaneuh** [comment]
Glad to see you back to business @ronanguilloux ! :-)

**Pull request merged**